logfire/_internal/integrations/redis.py

```diff
@@ -21,9 +25,32 @@ class RedisInstrumentKwargs(TypedDict, total=False):
         skip_dep_check: bool
 
 
-def instrument_redis(**kwargs: Unpack[RedisInstrumentKwargs]) -> None:
+def instrument_redis(capture_statement: bool = False, **kwargs: Unpack[RedisInstrumentKwargs]) -> None:
     """Instrument the `redis` module so that spans are automatically created for each operation.
 
     See the `Logfire.instrument_redis` method for details.
+
+    Args:
+        capture_statement: Whether to capture the statement being executed. Defaults to False.
+        **kwargs: Additional keyword arguments to pass to the `RedisInstrumentor.instrument` method.
     """
-    RedisInstrumentor().instrument(**kwargs)  # type: ignore[reportUnknownMemberType]
+    request_hook = kwargs.pop('request_hook', None)
+    if capture_statement:
+        request_hook = _capture_statement_hook(request_hook)
+
+    RedisInstrumentor().instrument(request_hook=request_hook, **kwargs)  # type: ignore[reportUnknownMemberType]
+
+
+def _capture_statement_hook(request_hook: RequestHook | None = None) -> RequestHook:
+    truncate_value = functools.partial(truncate_string, max_length=20, middle='...')
+
+    def _capture_statement(
+        span: Span, instance: Connection, command: tuple[object, ...], *args: Any, **kwargs: Any
+    ) -> None:
+        str_command = list(map(str, command))
+        span.set_attribute('db.statement', ' '.join(str_command))
```

Comment:
We should ensure that the frontend doesn't try to format these queries.

Reply:
<img width="1492" alt="Screenshot 2024-08-06 at 11 20 04" src="https://github.com/user-attachments/assets/c6ec9953-1b6b-4af1-9d22-2a5ea59d6923">
All good?